src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -678,4 +676,33 @@ object messages {
            |$fixedVarInAlternative""".stripMargin
     }
   }
+
+  case class IdentifierExpected()(implicit ctx: Context) extends Message(24) {
+    val kind = "Syntax"
+    val msg = "identifier expected"
+
+    val typeExplicitly = "def sum(): Int = {...}"
+
+    val typeImplicit = "def sum() = {...}"
+
+    val explanation = {
+      hl"""|When using type identifiers, you should pass a valid one.
+           |Because of Scala's stronger type, you can fix this issue either
+           |passing a valid type identifier or you can omit it, so the compiler can infer it for you.
+           |
+           |Let's write a function called ${"sum"}
+           |
+           |Here we explicitly pass a valid type identifier, in this case ${"Int"} for the function:
+           |
+           |$typeExplicitly
+           |
+           |But you also have the option to omit it completely.
+           |
+           |$typeImplicit
+           |
+           |This time the compiler will infer this type for you.
+           |
+           |""".stripMargin
+    }
+  }
```

Comment:
So I have a couple of comments/questions on this explanation:

* Does it make sense for all cases?
* Could you show the user their code and the correct way to rewrite it? E.g: `def foo: int = { ... }` => `def foo = { ... }`
* There are some weird things when it comes to the language like `"Because of Scala's stronger type"` - wat? Have a look through it and I'm sure you'll find places to correct :)
*  The line breaking is increasing, first line is something like 60 chars, the second 65 and third 90+, try to even it out.
* The line with "Here we explicitly" is too long (84 chars). We want a line to be max 80 chars (the standard terminal width).